easybuild/easyblocks/i/intel_compilers.py

```diff
@@ -0,0 +1,124 @@
+# #
+# Copyright 2021-2021 Ghent University
+#
+# This file is part of EasyBuild,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/easybuilders/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+# #
+"""
+EasyBuild support for installing Intel compilers, implemented as an easyblock
+
+@author: Kenneth Hoste (Ghent University)
+"""
+import os
+from distutils.version import LooseVersion
+
+from easybuild.easyblocks.generic.intelbase import IntelBase
+from easybuild.tools.build_log import EasyBuildError
+
+
+class EB_intel_minus_compilers(IntelBase):
+    """
+    Support for installing Intel compilers, starting with verion 2021.x (oneAPI)
+    """
+
+    def __init__(self, *args, **kwargs):
+        """
+        Easyblock constructor: check version
+        """
+        super(EB_intel_minus_compilers, self).__init__(*args, **kwargs)
+
+        # this easyblock is only valid for recent versions of the Intel compilers (2021.x, oneAPI)
+        if LooseVersion(self.version) < LooseVersion('2021'):
+            raise EasyBuildError("Invalid version %s, should be >= 2021.x" % self.version)
+
+        self.compilers_subdir = os.path.join('compiler', 'latest', 'linux')
+
+    def prepare_step(self, *args, **kwargs):
+        """
+        Prepare environment for installing.
+
+        Specify that oneAPI versions of Intel compilers don't require a runtime license.
+        """
+        # avoid that IntelBase trips over not having license info specified
+        kwargs['requires_runtime_license'] = False
+
+        super(EB_intel_minus_compilers, self).prepare_step(*args, **kwargs)
+
+    def configure_step(self):
+        """Configure installation: specify components to install."""
+
+        # redefine $HOME, to avoid that anything is stored in $HOME/intel (like the 'installercache' database)
+        os.environ['HOME'] = self.builddir
```

Comment:
This is problematic, it broke GitHub integration for me when there was a failure. It also doesn't correspond to what happens with `https://github.com/easybuilders/easybuild-easyblocks/pull/2305/files#diff-3dc6b8bfa78867742693cea33fdba081128e823f26ba25e177dd68d83adfec62R211`, this should be used instead?

Reply:
If not then you will also need a custom cleanup step, or make the one in intelbase a bit more resilient

Reply:
I think you should revert the redefinition of `HOME` as soon as you no longer need it, and make `clean_home_subdir()` only do something if the directory actually exists. That should fix the issues I saw.

Reply:
FYI: The installer team confirmed that setting the HOME variable is the best way to control where the files are dropped.

Reply:
Thanks for the feedback @rscohn2!

Reply:
What about changing this line to
```
self.cfg['preinstallopts'] += " HOME=%s " % self.builddir
```